components/server/src/server.rs

```diff
@@ -1433,28 +1351,36 @@ impl TiKVServer<RaftLogEngine> {
             dump_raftdb_to_raft_engine(&raftdb, &raft_engine, 8 /*threads*/);
             raft_data_state_machine.after_dump_data();
         }
+        raft_engine
+    }
+}
+
+impl<Er: ConfiguredRaftEngine> TiKVServer<Er> {
+    fn init_raw_engines(
+        &mut self,
+        flow_listener: engine_rocks::FlowListener,
+    ) -> (Engines<RocksEngine, Er>, Arc<EnginesResourceInfo>) {
+        let block_cache = self.config.storage.block_cache.build_shared_cache();
+        let env = self
+            .config
+            .build_shared_rocks_env(self.encryption_key_manager.clone(), get_io_rate_limiter())
+            .unwrap();
+
+        let raft_engine = Er::build(self, &env, &block_cache);
```

Comment:
how about adding a struct called `RaftEngineFactory` and use same manner as `KvEngineFactory`

Reply:
They are not the same. `KvEngineFactory` will be reused in the future when creating multiple rocksdb inside a TiKV process.

Reply:
Oh, RaftEngine is shared

Reply:
but `KvEngineFactory` should still focus on building a kv engine, we'd make them in the same manner

Reply:
It seems over abstracted to me giving that it will not be used again in anywhere.

Reply:
what I expected is 
```
fn init_raw_engines() {
    let env = xxx;
    let cfg_controller = xx;
    let engines = Engines::default();

    let raft_engine = RaftEngineBuilder::build();
    raft_engine.register_config(&cfg_controller);
    engines.raft = raft_engines;

    for tablet in tablets {
       let kv_engine = KvEngineBuilder::build();
       kv_engine.register_config(&cfg_controller);
       engines.push_kv(kv_engine);
    }

    engines
}
```

Reply:
`RaftEngineFactory` doesn't introduce additional abstract, it is just a rewrite of `ConfiguredRaftEngine`

Reply:
So to implement `RaftEngineBuilder`, there has to be another trait `RaftEngineBuilder` and then implement `RaftEngineBuilder` to `RaftLogEngineBuilder` and `RocksEngineAsRaftEngineBuilder`.

To implement `register_config`, there should be an extra trait `RegisterConfigTrait` and implement it to RocksEngine and RaftLogEngine.

So there will be two traits and two other types compared to current implementation, which only requires only one trait.